testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -598,6 +598,13 @@ void testCastBooleanToNumeric(CastType castType, SqlOperatorFixture f) {
         "Cast function cannot convert value of type BOOLEAN to type DECIMAL\\(19, 0\\)", false);
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/projects/CALCITE/issues/CALCITE-6322">
+   *  [CALCITE-6322] Casts to DECIMAL types are ignored</a>. */
```

Comment:
The PR claims to add support for three kinds of CASTS:
* decimal to decimal
* int to decimal
* double/float to decimal

I've seen a few tests for the first category, do we have something for the other two? If not then I think we should add.

Moreover, I've noticed some tests that are currently disabled by the `DECIMAL` flag. A few of them should be fixed by this PR so please enable those that are relevant.

Reply:
I will re-enable many of these tests, but this doesn't completely solve the cases covered by SqlOperatorTest.DECIMAL. I have filed a new issue https://issues.apache.org/jira/browse/CALCITE-6379 that covers arithmetic. Moreover, the current PR also does not cover casts from STRING and INTERVAL. I have filed  https://issues.apache.org/jira/browse/CALCITE-6380 to cover these two cases.